databricks/koalas/namespace.py

```diff
@@ -1863,6 +1863,85 @@ def merge(obj, right: 'DataFrame', how: str = 'inner',
         left_index=left_index, right_index=right_index, suffixes=suffixes)
 
 
+def to_numeric(arg):
+    """
+    Convert argument to a numeric type.
+
+    Parameters
+    ----------
+    arg : scalar, list, tuple, 1-d array, or Series
+
+    Returns
+    -------
+    ret : numeric if parsing succeeded.
+
+    See Also
+    --------
+    DataFrame.astype : Cast argument to a specified dtype.
+    to_datetime : Convert argument to datetime.
+    to_timedelta : Convert argument to timedelta.
+    numpy.ndarray.astype : Cast a numpy array to a specified type.
+
+    Examples
+    --------
+
+    >>> kser = ks.Series(['1.0', '2', '-3'])
+    >>> kser
+    0    1.0
+    1      2
+    2     -3
+    Name: 0, dtype: object
+
+    >>> ks.to_numeric(kser)
+    0    1.0
+    1    2.0
+    2   -3.0
+    Name: 0, dtype: float32
+
+    If given Series contains invalid value to cast float, just cast it to `np.nan`
+
+    >>> kser = ks.Series(['apple', '1.0', '2', '-3'])
+    >>> kser
+    0    apple
+    1      1.0
+    2        2
+    3       -3
+    Name: 0, dtype: object
+
+    >>> ks.to_numeric(kser)
+    0    NaN
+    1    1.0
+    2    2.0
+    3   -3.0
+    Name: 0, dtype: float32
+
+    Also support for list, tuple, np.array, or a scalar
+
+    >>> ks.to_numeric(['1.0', '2', '-3'])
+    array([ 1.,  2., -3.])
+
+    >>> ks.to_numeric(('1.0', '2', '-3'))
+    array([ 1.,  2., -3.])
+
+    >>> ks.to_numeric(np.array(['1.0', '2', '-3']))
+    array([ 1.,  2., -3.])
+
+    >>> ks.to_numeric('1.0')
+    1.0
+    """
+    if isinstance(arg, Series):
+        sdf = arg._internal.sdf
+        name = arg.name
+        sdf = sdf.withColumn(name, sdf[name].cast('float'))
```

Comment:
I think we shouldn't always cast it to `float`. We can:

1. Disallow `errors='ignore'` case because it looks this is the only case `object` type is allowed.
2. Infer schema from the small sample (see `Series.apply` as an example)
3. Cast it to the inferred type as you did. Alternatively we can also use `pandas_udf` to allow to throw an exception by given `errors` parameters.

Reply:
What's the conclusion?
Seems like we just cast to float so far?

Reply:
If we just simply cast to float, then we should:

```py
return arg._with_new_scol(arg._internal.scol.cast('float'))
```

?